pkgs/by-name/st/stdoutisatty/package.nix

```diff
@@ -0,0 +1,70 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  makeBinaryWrapper,
+  nix-update-script,
+  runCommand,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "stdoutisatty";
+  version = "1.0";
+
+  src = fetchFromGitHub {
+    owner = "lilydjwg";
+    repo = finalAttrs.pname;
```

Comment:
This is discouraged: https://github.com/NixOS/nixpkgs/issues/310373

Reply:
Thank you for linking this interesting discussion! I didn't know. I have removed most superfluous use of `finalAttrs`. In particular, all fields of `src` are now spelled out explicitly. However, it does make me a bit uncomfortable to either:
- specify twice the version `"1.0"` which should clearly be identified, or
- use `rec` :laughing: 

So I opted for a slightly ugly fix: instead of defining `src.rev` with `version`, define `version` with `src.rev`:
```nix
  version = finalAttrs.src.rev;

  src = fetchFromGitHub {
    owner = "lilydjwg";
    repo = "stdoutisatty";
    rev = "1.0";
    hash = "sha256-NyVn9cxx0rY1ridNDTqe0pzcVhdLVaPCKT4hoQkQzRs=";
  };
```
Is it okay to do this? :laughing:

Reply:
Actually it's fine wot use finalAttrs.version in rev. Just avoid pname is enough.